### ISSUE TITLE
Add support for shepherd (GuixSD)

### ISSUE THREAD

**Issue by Ambrevar** [cbowdon/daemons.el#7]
I would be fantastic to see support for [Shepherd](https://www.gnu.org/software/shepherd/), the GNU init system (used, among others, by [GuixSD](https://www.gnu.org/software/guix/)).

**Comment by cbowdon** [comment]
@jellelicht has kindly provided a PR for this, which is now merged in. Please give it a whirl!

**Comment by Ambrevar** [comment]
Off to a rough start: it does not load :p 

Fix:
``` diff
-        ((= 0 (daemons--shell-command "which herd")) 'daemons-herd)
+        ((= 0 (daemons--shell-command "which herd")) 'daemons-shepherd)
```

Once past that, `daemons` does not display anything, I still haven't figured out why.

**Comment by Ambrevar** [comment]
@jellelicht?

**Comment by Ambrevar** [comment]
My bad: `daemons-always-sudo` needs to be non-nil for daemons-shepherd to work.

**Comment by cbowdon** [comment]
Oops, that was a lousy mistake. Fixed in 40f8e5a6a4891a972e3b003f7f4a39667c63edaf.

**Comment by Ambrevar** [comment]
Thanks!

**Comment by Ambrevar** [comment]
Few hints:

- Get the TRAMP host with `(tramp-file-name-host (tramp-dissect-file-name ...))`.

- Cache your saved data in a structure stored in a hash-map / alist indexed by host-name.

**Comment by jellelicht** [comment]
I only wrote (and tested) for Shepherd /w user services, as mentioned in the README. If it does not work for that, I did something wrong. I also kind of assumed that there would already be a running Shepherd instance running *somewhere*, if this is incorrect I should work on that.

AFAICS, you are trying to use system shepherd, which I did not test for. Is there something I can still do to make it work for your use-case as well?
